test/unit/spec/common/cellComponents/paramsWidget-Spec.js

```diff
@@ -0,0 +1,92 @@
+/*global describe, it, expect*/
+/*global beforeEach, beforeAll, afterAll*/
+/*jslint white: true*/
+
+define([
+    'common/cellComponents/paramsWidget',
+    'common/runtime',
+    'common/props',
+    'common/spec',
+    'json!../../../../data/testAppObj.json',
+], (ParamsWidget, Runtime, Props, Spec, TestAppObject) => {
+    'use strict';
+
+    describe('The Parameter module', () => {
+        it('loads', () => {
+            expect(ParamsWidget).not.toBe(null);
+        });
+
+        it('has expected functions', () => {
+            expect(ParamsWidget.make).toBeDefined();
+        });
+    });
+
+    describe('The Parameter instance', () => {
+        beforeAll(() => {
+            Jupyter.narrative = {
+                getAuthToken: () => 'fakeToken',
+            };
+        });
+
+        afterAll(() => {
+            Jupyter.narrative = null;
+        });
+
+        let paramsWidget, node, spec, parameters;
+
+        beforeEach(() => {
+            const bus = Runtime.make().bus();
+            node = document.createElement('div');
+            document.getElementsByTagName('body')[0].appendChild(node);
+
+            const model = Props.make({
+                data: TestAppObject,
+                onUpdate: (props) => {},
+            });
+
+            spec = Spec.make({
+                appSpec: model.getItem('app.spec'),
+            });
+
+            parameters = spec.getSpec().parameters;
+
+            const workspaceId = 54745;
+
+            paramsWidget = ParamsWidget.make({
+                bus: bus,
+                workspaceId: workspaceId,
+                initialParams: model.getItem('params'),
+            });
+        });
+
+        it('should start and render itself', () => {
+            return paramsWidget
+                .start({
+                    node: node,
+                    appSpec: spec,
+                    parameters: parameters,
+                })
+                .then(() => {
+                    expect(node.innerHTML).toContain('Parameters');
```

Comment:
I would also check for specific parameters to make sure they were rendered too

Reply:
You can import jQuery / use `document.querySelector` and then use it (1) to ensure that the resulting structure contains the correct classes, etc., and (2) that the elements have the correct content.

You can also do a `element.click()` test to ensure that the show/hide advanced params thing works, and that it is set in the correct state when the params widget is created.